### DISCUSSION TITLE
dynamic class generation not working with width custom size percentage

### DISCUSSION THREAD

**Discussion by DelrosarioRovic** [tailwindlabs/tailwindcss#12946] [Help] [answered]
I was doing a react vite + tailwind 

and i just encounter a design that did not working

![image](https://github.com/tailwindlabs/tailwindcss/assets/118983530/329042d4-136f-431c-a6eb-4729e73c6729)

when i did this its not gonna work but sometimes it was working 

like when i change the number to 50, 80, 90, 40

but its not working between 51 - 69 
also with the percent with non-zero ending

please help

**Comment by wongjn** [comment, marked as answer]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

Consider using the `style` attribute instead, like:

```jsx
<div … style={{ width: `${70}%` }}>
```

**Reply by DelrosarioRovic** [reply]
thanks that's help me a lot

**Comment by jaimeenkoffeekodes** [comment]
just use this very small library : **npm i tailwind-merge** , it will make much more easy task with conditional rendering of tailwind classes.
